tests/crypto_policy_test.go

```diff
@@ -0,0 +1,235 @@
+package tests
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"net"
+	"net/http"
+	"strconv"
+
+	. "github.com/onsi/ginkgo"
+	"github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+	ocpv1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/library-go/pkg/crypto"
+	core "k8s.io/api/core/v1"
+	apiregv1 "k8s.io/kube-aggregator/pkg/apis/apiregistration/v1"
+	sspv1beta1 "kubevirt.io/ssp-operator/api/v1beta1"
+	"kubevirt.io/ssp-operator/internal/common"
+	"kubevirt.io/ssp-operator/tests/env"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+)
+
+var _ = Describe("Crypto Policy", func() {
+	BeforeEach(func() {
+		waitUntilDeployed()
+	})
+
+	Context("setting Crypto Policy", func() {
+		sspPod := operatorPod()
+		table.DescribeTable("Adhere to defined TLSConfig", testSettingTLSConfig,
+			table.Entry("[test_id:???] old", sspPod, oldPermutation),
+			table.Entry("[test_id:???] intermediate", sspPod, intermediatePermutation),
+			table.Entry("[test_id:???] modern", sspPod, modernPermutation),
+			table.Entry("[test_id:???] custom", sspPod, customPermutation),
+		)
+	})
+})
+
+func operatorPod() core.Pod {
+	cfg, err := config.GetConfig()
+	Expect(err).ToNot(HaveOccurred())
+	apiClient, err = client.New(cfg, client.Options{Scheme: testScheme})
+	Expect(err).ToNot(HaveOccurred())
+	pods := &core.PodList{}
+	err = apiClient.List(context.TODO(), pods, client.MatchingLabels{"control-plane": "ssp-operator"})
+	Expect(err).ToNot(HaveOccurred())
+	Expect(pods.Items).ToNot(BeEmpty())
+	return pods.Items[0]
+}
+
+type tlsConfigTestPermutation struct {
+	openshiftTLSPolicy *ocpv1.TLSSecurityProfile
+	allowedConfigs     []clientTLSOptions
+	disallowedConfigs  []clientTLSOptions
+}
+
+type clientTLSOptions struct {
+	OpenSSLCipherNames []string
+	MaxTLSVersion      uint16
+}
+
+func (s *clientTLSOptions) CipherIDs() (cipherSuites []uint16) {
```

Comment:
These (`crypto.OpenSSLToIANACipherSuites` and `common.GetKnownCipherId`) are functions of the code to be tested. Can we fix the values somehow so the test code and tested code are not dependent on each other?